### PR TITLE
docs: create testing strategy

### PR DESCRIPTION
**Pull request by sarahndyson**
# Description

Document Testing Strategy for Enhance Authentication

## Type of change

- [x] Documentation (update or new)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [x] I have requested a review from @talzid and @kats8 on the Pull Request

### CODE REVIEW

**Review by kats8 — COMMENTED** [review]
Thanks, @sarahndyson 

This is looking pretty good :) but I would suggest your test strategy should include “quality assurance” deliverables that would be produced - in this case, that is likely to include a testing plan: your testing plan will outline different testing scenarios and their expected testing result (as well as being used to track actual results). You don’t need to define these scenarios within your testing strategy itself, but you can simply mention it (testing plan) as a deliverable to be produced. Testing scenarios/expected results should then reflect requirements of your product.

Note: there is a simple testing plan Markdown [template](https://github.com/thoth-tech/handbook/blob/main/docs/processes/quality-assurance/templates/test-plan-template.md) in the handbook.

Keep up the good work!

File: docs/OnTrack/Deployment/Enhanced Authentication/testing-strategy-enhance-authentication.md
Comment (talzid) [inline docs/OnTrack/Deployment/Enhanced Authentication/testing-strategy-enhance-authentication.md:7]: I would put the references as the bottom of the document - almost as an appendix. It isn't 100% required knowledge for the testing strategy but it is useful to have.

**Review by talzid — COMMENTED** [review]
Thank you for the excellent work @sarahndyson, this is a great starting point for some info about the testing strat.

Once you've added some discussion around the testing plan and cases this will be ready to merge. @kats8 is here to help / partner you with someone who will help you expand this document.

File: docs/OnTrack/Deployment/Enhanced Authentication/testing-strategy-enhance-authentication.md
Comment (kats8) [reply docs/OnTrack/Deployment/Enhanced Authentication/testing-strategy-enhance-authentication.md:7]: You'll have to blame me for this ordering, @talzid - it follows the order that comes from our handbook testing strategy template (based on a format from _Programming Foundations: Software Testing/QA from LinkedIn Learning_).

Comment (talzid) [reply docs/OnTrack/Deployment/Enhanced Authentication/testing-strategy-enhance-authentication.md:7]: Nevermind my comment then. If Kat says it is, then that it shall be :smile: I trust the method and research

**Comment by sarahndyson** [comment]
Thank you so much @kats8 and @talzid for the feedback!
I've updated the testing strategy - please let me know if any further improvements can be made :)

**Comment by talzid** [comment]
Looks good to me. I imagine in future there will be a test case and management strategy but this is a good outline of the plan.
If @kats8 agrees, I am happy to merge as is and iterate on this when new test case stuff is created!

Before it can be merged, please use the prettier extension to format the file in VS code. This will allow it to satisfy the build check it is currently failing. If you need help with this, please reach out

**Review by kats8 — APPROVED** [review]
Thanks, @sarahndyson. Those updates look good 👍  Happy for this to be merged :).

**Comment by sarahndyson** [comment]
Hi again @talzid, I managed to get prettier up and running so that the checks are all passed now so I believe that this is ready to merge :)

**Comment by talzid** [comment]
Perfect!

**Pull request merged**
